checker/src/main/java/org/checkerframework/checker/resourceleak/MustCallInferenceLogic.java

```diff
@@ -40,107 +90,674 @@ public class MustCallInferenceLogic {
   /** The {@link Owning} annotation. */
   protected final AnnotationMirror OWNING;
 
-  /** The control flow graph. */
+  /**
+   * The control flow graph of the current method. There is a separate MustCallInferenceLogic for
+   * each method.
+   */
   private final ControlFlowGraph cfg;
 
+  /** The MustCallConsistencyAnalyzer. */
+  private final MustCallConsistencyAnalyzer mcca;
+
+  /** The MethodTree of the current method. */
+  private final MethodTree enclosingMethodTree;
```

Comment:
It's pointing to the method being analyzed. Since we are analyzing its body and statements, I used `enclosing`, but I think `currentMethodTree` or just `methodTree` is a better option.